ops/main.py

```diff
@@ -42,6 +43,38 @@
 logger = logging.getLogger()
 
 
+@dataclasses.dataclass(frozen=True)
+class _JujuContext:
+    # source: https://juju.is/docs/juju/charm-environment-variables
+    # the HookVars function: https://github.com/juju/juju/blob/3.6/worker/uniter/runner/context/context.go#L1398
+    # only allow accesing these vars because these are the currently used vars in main.py
+
+    juju_relation: str = ''
```

Comment:
I think this should be called `relation_name` for clarity, even though the env var is `JUJU_RELATION`.

Also, I think Tony mentioned it already somewhere, but let's drop the `juju_` prefix, as that's true of all of them (and this is the "juju context" object).